projects/ng2-smart-table/src/lib/lib/grid.ts

```diff
@@ -328,4 +334,9 @@ export class Grid {
     const maxPageAmount: number = Math.ceil(source.count() / perPage);
     return maxPageAmount ? Math.min(pageToSelect, maxPageAmount) : pageToSelect;
   }
+
+  private isNeedToSelectRow(): boolean {
+    const selectedRowIndex = Number(this.getSetting('selectedRowIndex'));
+    return !selectedRowIndex || selectedRowIndex > 0;
```

Comment:
`Number(...)` can return only number or Nan,
`Nan >= 0` always `false`
Maybe `!selectedRowIndex` check isn't necessary?

Reply:
we do not need to select row in one case - when selectedRowIndex < 0. in other cases - selectedRowIndex === undefined | null | 'asd' | >= 0 - we need to select row